### PR TITLE
feat(utils): add a lodash-compatible `get` function

### LINKED ISSUE #1056: Add a lodash-compatible `get` function to the `@tonic-ui/utils` package

### PR DESCRIPTION
**Pull request by cheton**
Package | Release | Reason
:-- | :-- | :-
@tonic-ui/mcp | patch | Internal updates, no new user-facing features
@tonic-ui/react | patch | Internal updates, no new user-facing features
@tonic-ui/react-base | patch | Internal updates, no new user-facing features
@tonic-ui/styled-system | patch | Now uses `get` from `@tonic-ui/utils`; behavior is backward-compatible
@tonic-ui/utils | minor | Adds lodash-compatible `get` function with bracket notation support (new feature)

### **PR Type**
Enhancement, Tests

___

### **Description**
- Add lodash-compatible `get` utility function with path parsing

- Fix React testing warnings with proper `act()` wrapping

- Add missing build scripts to package configuration

- Improve test consistency with default values

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["New get() function"] --> B["Path parsing logic"]
  B --> C["Object property access"]
  D["Test improvements"] --> E["act() wrapper"]
  D --> F["Default values"]
  G["Package scripts"] --> H["pre-push & prepublish"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>shared.js</strong><dd><code>Add lodash-compatible get utility function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/utils/src/shared.js

<ul><li>Add <code>get</code> function with lodash-compatible API<br> <li> Implement <code>_parsePath</code> helper for dot/bracket notation<br> <li> Support arrays, objects, and nested property access<br> <li> Handle edge cases with null/undefined values</ul>

</details>

  </td>
  <td><a href="https://github.com/trendmicro-frontend/tonic-ui/pull/1057/files#diff-aeb582188d4c096beb54627221b5a143210c728a566d8328830e6d31e1fee116">+65/-15</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>shared.test.js</strong><dd><code>Add comprehensive tests for get function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/utils/src/__tests__/shared.test.js

<ul><li>Add comprehensive test suite for <code>get</code> function<br> <li> Test compatibility with lodash implementation<br> <li> Cover edge cases, arrays, objects, and special characters<br> <li> Import lodash get for comparison testing</ul>

</details>

  </td>
  <td><a href="https://github.com/trendmicro-frontend/tonic-ui/pull/1057/files#diff-d141d14bbe6649708f67ab3f9f67f00741331be53b2f0be2fa5335b2e2e9ad97">+121/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>DatePicker.test.js</strong><dd><code>Fix focus call with act wrapper</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/react/src/date-pickers/DatePicker/__tests__/DatePicker.test.js

<ul><li>Import <code>act</code> from React Testing Library<br> <li> Wrap <code>datePickerInput.focus()</code> call in <code>act()</code><br> <li> Fix React testing warnings for DOM updates</ul>

</details>

  </td>
  <td><a href="https://github.com/trendmicro-frontend/tonic-ui/pull/1057/files#diff-162359f47cac8420632aa4c1bda1d0b190a69128c61126ac42c2c29f6441d8fc">+4/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>RadioGroup.test.js</strong><dd><code>Add default values to RadioGroup tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/react/src/radio/__tests__/RadioGroup.test.js

<ul><li>Add <code>defaultValue=""</code> prop to all RadioGroup components<br> <li> Ensure consistent test setup across test cases<br> <li> Apply to string, number, boolean, object, and disabled tests</ul>

</details>

  </td>
  <td><a href="https://github.com/trendmicro-frontend/tonic-ui/pull/1057/files#diff-dd95ea46d39722fe882e318cd802d618b836e92b572ceacf5c7f28db932f87de">+5/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.test.js</strong><dd><code>Update exports test for get function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/utils/__tests__/index.test.js

- Add `get` to expected exports list
- Update export validation test

</details>

  </td>
  <td><a href="https://github.com/trendmicro-frontend/tonic-ui/pull/1057/files#diff-c93f9ef14e4b40fb5601e0e75a3330dbf8f8e9ab9b10d7669666e3c9f3d8e31f">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.test.js</strong><dd><code>Update exports test for get function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/utils/src/__tests__/index.test.js

- Add `get` to expected exports list
- Update export validation test

</details>

  </td>
  <td><a href="https://github.com/trendmicro-frontend/tonic-ui/pull/1057/files#diff-1767f4b6010d53f0a7d1e157e178d6127602e4d45ed7c8b6dc686a25ce36229a">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Add missing package scripts</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/mcp/package.json

<ul><li>Add <code>pre-push</code> script with build, lint, and test commands<br> <li> Add <code>prepublish</code> script to run build before publishing<br> <li> Remove unused <code>dev</code> and <code>start</code> scripts</ul>

</details>

  </td>
  <td><a href="https://github.com/trendmicro-frontend/tonic-ui/pull/1057/files#diff-adc96623016293ce329af7c2c6419b527d44d978a2ba059b926a85678439a3db">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: .changeset/tonic-ui-1057.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+"@tonic-ui/mcp": patch
+"@tonic-ui/react": patch
+"@tonic-ui/react-base": patch
+"@tonic-ui/styled-system": patch
+"@tonic-ui/utils": minor
+---
```
Comment (cheton) [inline .changeset/tonic-ui-1057.md:7]: Package | Release | Reason
:-- | :-- | :-
@tonic-ui/mcp | patch | Internal updates, no new user-facing features
@tonic-ui/react | patch | Internal updates, no new user-facing features
@tonic-ui/react-base | patch | Internal updates, no new user-facing features
@tonic-ui/styled-system | patch | Now uses `get` from `@tonic-ui/utils`; behavior is backward-compatible
@tonic-ui/utils | minor | Adds lodash-compatible `get` function with bracket notation support (new feature)
